Composer/packages/client/src/pages/botProject/adapters/ABSChannels.tsx

```diff
@@ -122,27 +123,40 @@ export const ABSChannels: React.FC<RuntimeSettingsProps> = (props) => {
       TelemetryClient.track('ConnectionsAddNewProfile');
       navigateTo(`/bot/${projectId}/publish/all/#addNewPublishProfile`);
     } else {
-      let newtoken = '';
-      if (userShouldProvideTokens()) {
-        if (isShowAuthDialog(false)) {
-          setShowAuthDialog(true);
-        }
-        newtoken = getTokenFromCache('accessToken');
-      } else {
-        newtoken = await AuthClient.getAccessToken(armScopes);
-      }
-      setToken(newtoken);
-
       // identify the publishing profile in the list
       const profile = publishTargets?.find((p) => p.name === opt.key);
       if (profile) {
         const config = JSON.parse(profile.configuration);
+
         setCurrentResource({
           microsoftAppId: config?.settings?.MicrosoftAppId,
           resourceName: config.botName || config.name,
           resourceGroupName: config.resourceGroup || config.botName || config.name,
+          tenantId: config.tenantId,
           subscriptionId: config.subscriptionId,
         });
+
+        let newtoken = '';
+        if (userShouldProvideTokens()) {
+          if (isShowAuthDialog(false)) {
+            setShowAuthDialog(true);
+          }
+          newtoken = getTokenFromCache('accessToken');
+        } else {
+          try {
+            // if tenantId is present, use this to retrieve the arm token.
+            // absence of a tenantId indicates this was a legacy (pre-tenant support) provisioning profile
+            if (config.tenantId) {
+              newtoken = await AuthClient.getARMTokenForTenant(config.tenantId);
+            } else {
+              newtoken = await AuthClient.getAccessToken(armScopes);
```

Comment:
This is still a broken path (which may be fine for this release). I think we could follow the same pattern in [Publish.tsx](https://github.com/microsoft/BotFramework-Composer/blob/main/Composer/packages/client/src/pages/publish/Publish.tsx#L303) and show a error notification which prompts the user to edit their configuration.

Reply:
But wouldn't it only be broken if the user had *successfully* created a publishing profile in the past using the default tenant used when accessing the getAccessToken method.... right?

Reply:
This is just broken for users with MSA accounts I think. Just saying we have an opportunity to address that here and should consider doing it because is addresses the root cause of the original issue.

Reply:
I added these additional checks to try to fetch a tenantId from the tenant list, etc.